### PR TITLE
Fix boolean conversion issues, add regression test, fix indentation

### LINKED ISSUE #261: Bug with nullable bool
Hello.

My entity contains nullable bool field.

```dart
@ColumnInfo(nullable: true)
final bool visa;
```

When I call this method with visa == null:

```dart
@Insert(onConflict: OnConflictStrategy.REPLACE)
Future<int> insertOrUpdate(Place place);
```

I got crash:

image

image

### PR DESCRIPTION
**Pull request by mqus**
This fixes #261, along with multiple other bool-to-int-to-bool conversion issues involving null and the `?` substitution made by sqflite.
I added the regression tests as integration tests because this is a behaviour which very much depends on sqflite. Even if these issues might be fixed by the TypeConverters soon, it might be good to at least keep the tests.

### CODE REVIEW

File: floor_generator/lib/writer/query_method_writer.dart [hunk 1]
```diff
@@ -94,7 +94,11 @@ class QueryMethodWriter implements Writer {
     return _queryMethod.parameters
         .map((parameter) {
           if (!parameter.type.isDartCoreList) {
-            return parameter.displayName;
+            if (parameter.type.isDartCoreBool) {
+              return '${parameter.displayName} == null ? null : ${parameter.displayName} ? 1 : 0';
```
Comment (vitusortner) [inline floor_generator/lib/writer/query_method_writer.dart:98]: ⚠️This mapping also is quite advanced and not very readable. Let's talk about how we cold simplify that.

Please add some test for this new functionality.

File: floor_generator/lib/value_object/entity.dart [hunk 2]
```diff
@@ -57,19 +57,25 @@ class Entity extends Queryable {
...
-  String _getAttributeValue(final FieldElement fieldElement) {
-    final parameterName = fieldElement.displayName;
-    return fieldElement.type.isDartCoreBool
-        ? 'item.$parameterName ? 1 : 0'
-        : 'item.$parameterName';
+  String _getAttributeValue(final Field field) {
+    final parameterName = field.fieldElement.displayName;
+    if (field.fieldElement.type.isDartCoreBool) {
+      if (field.isNullable) {
+        return 'item.$parameterName == null ? null : item.$parameterName ? 1 : 0';
```
[the remark this answers is not in the document]
Comment (mqus) [reply floor_generator/lib/value_object/entity.dart:72]: I know a similar function from Rust, where you can 
```rust
nullablebool.map(|notnullbool| if notnullbool {1} else {0}) `
```
But i know no such function in dart... I don't even know if you can define an extension on `bool`... 

I tried playing with 
```dart
variable?.
```
but it only seems to allow fields and not inclusion in arbitrary expressions. There is also https://pub.dev/packages/optional which somewhat implements the Rust way but I think this is too much here. 

I think the only sensible approach I know is to use brackets/parentheses.

File: floor_generator/lib/processor/queryable_processor.dart [hunk 3]
```diff
@@ -80,7 +80,7 @@ abstract class QueryableProcessor<T extends Queryable> extends Processor<T> {
     final String parameterValue,
   ) {
     if (parameterType.isDartCoreBool) {
-      return '($parameterValue as int) != 0'; // maps int to bool
+      return '$parameterValue == null ? null : ($parameterValue as int) != 0'; // maps int to bool
```
[the remark this answers is not in the document]
Comment (mqus) [reply floor_generator/lib/processor/queryable_processor.dart:83]: Yeah the complexity was a reason, as well as my lazyness :) I can change that here if you want.

[the remark this answers is not in the document]
Comment (mqus) [reply floor_generator/lib/processor/queryable_processor.dart]: I added the non-nullable bool case now, along with a test which should cover both cases

File: floor_generator/lib/value_object/entity.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (mqus) [reply floor_generator/lib/value_object/entity.dart]: I added the parentheses now. I'm not sure if this is that important if it's only generated code...

File: floor_generator/lib/writer/query_method_writer.dart (hunk 1 above)
Comment (mqus) [reply floor_generator/lib/writer/query_method_writer.dart]: Added tests and parentheses like already mentioned.

File: floor_generator/lib/value_object/entity.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (vitusortner) [reply floor_generator/lib/value_object/entity.dart]: I should've mentioned that the code snippet from above is Dart code that compiles.
The solution you came up with looks good. Let's keep it as it is 👍

File: floor_generator/lib/processor/queryable_processor.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (vitusortner) [reply floor_generator/lib/processor/queryable_processor.dart]: Thanks! 💯

**Pull request merged**